lib/AST/ProtocolConformance.cpp

```diff
@@ -149,8 +143,15 @@ ProtocolConformanceRef::getTypeWitnessByName(Type type,
   if (!assocType)
     return nullptr;
 
-  if (conformance.isAbstract())
+  if (conformance.isAbstract()) {
+    // For an archetype, retrieve the nested type with the appropriate
+    // name. There are no conformance tables.
+    if (auto archetype = type->getAs<ArchetypeType>()) {
```

Comment:
I... don't quite understand why moving this fixes the test case? I assume it means that we hit the `return nullptr`, but why are there look-ups for nested types that aren't associated types of the conformance's protocol in that test case?

Reply:
In our case, we have an archetype for which the conformance is concrete: it's an inherited conformance (since T inherits the class that conforms), and it can answer the "getTypeWitness" request.

Reply:
Oh, right, this also moved inside the `isAbstract` guard. Poor marks for reading comprehension.